myii/ssf-formula
# feat: overhaul CI implementation and use latest pre-salted images

Establish all of the following changes:

* Use wildcards for (but *maintain* current ordering of entries when rendered):
  - NOTE: Use new values to ensure reverse compatibility,
          where need to be able to use the original, non-wildcard method instead
  - To filter out `platforms` that must not be included (and `platform_matrix`)
  - `platform_suite` inclusions and exclusions
  - `platforms_matrix` (after initial testing)
* Use latest pre-salted images
  - Active unique combination of platforms for all formulas:
    + E.g. `apt-formula` must only have Debian-based platforms in the list
    + Based around idea of supplying these in the `FORMULA` file
    + Active became the wildcards that have been implemented
* Move `dependencies` to the top in each suite (where used)
* Use new solution to replace `platforms_matrix_commented_includes`:
  - Use wildcards?
  - Automatic generation from "ideal, unique platforms"?

When running global tests across new pre-salted images:

* `vim-interestingwords` is exceptionally useful here, with the following words:
  - `platforms` (use wildcards to ensure all relevant platforms are selected)
  - `supports` (pretty much want to comment all of these out, so InSpec tests run)
  - `include` & `exclude` (similar to `platforms`)
